### ISSUE TITLE
Large .xlsx file not getting .Sheets

### ISSUE THREAD

**Issue by jensJJ** [SheetJS/js-xlsx#792]
Hello,
I am using the newest library of `xlsx.js` on the client-side (browser). It works reading `xlsx` files and rendering to HTML table for smaller files.

Now I have a larger `xlsx` file with **340k** rows. I am getting the `sheetNames` attributes, but I am not getting the `.Sheets `object. It is empty.

What could be the error?
Thank you very much!

![excel1](https://user-images.githubusercontent.com/4704155/29668227-b0d75c4c-88df-11e7-813c-e1a3c8fd1551.png)

**Comment by sigod** [comment]
I have the same problem. I'm trying to process [XLSX file][0] (179772 rows, 38.1 MB) and `.Sheets` are just empty.

[0]: https://platform.prisma-capacity.eu/rest/reports/2329414/downloadReport

**Comment by SheetJSDev** [comment]
@sigod it's surpassing a JS engine limit.  XLSX files are ZIP containers, and the worksheet data is stored as XML text files within the ZIP.  Rename to .zip and extract and you'll see the file:

```
$ unzip -l 20160503_Auction_Report.xlsx
Archive:  /tmp/20160503_Auction_Report.xlsx
  Length      Date    Time    Name
---------  ---------- -----   ----
      592  05-03-2016 16:03   _rels/.rels
     1064  05-03-2016 16:03   [Content_Types].xml
      183  05-03-2016 16:03   docProps/app.xml
      438  05-03-2016 16:03   docProps/core.xml
  4245649  05-03-2016 16:03   xl/sharedStrings.xml
     2164  05-03-2016 16:03   xl/styles.xml
      357  05-03-2016 16:03   xl/workbook.xml
      570  05-03-2016 16:03   xl/_rels/workbook.xml.rels
392759204  05-03-2016 16:03   xl/worksheets/sheet1.xml <--
---------                     -------
397010221                     9 files
```

The current V8 string size limit (which applies to node and chrome) is `(1<<28) - 16 = 268435440` bytes and the file is significantly larger than that.

The good news is that the limit is being increased in the future: https://v8project.blogspot.com/2017/09/v8-release-62.html

**Comment by sigod** [comment]
@SheetJSDev Is there any good way to work around this issue?

**Comment by rupesh-pandey** [comment]
Yes any work around for this because I can see the data in Strings and able to access it manually without using API

**Comment by idibidiart** [comment]
Even with WTF option, I get no errors and it makes up "Sheet1" where the file actually contains many sheets none of which is named Sheet1.

File(265892023) {name: "Contract_Insight_Report_marzatem_JPMC.xlsm", lastModified: 1520893001000, lastModifiedDate: Mon Mar 12 2018 15:16:41 GMT-0700 (PDT), webkitRelativePath: "", size: 265892023, …}lastModified: 1520893001000lastModifiedDate: Mon Mar 12 2018 15:16:41 GMT-0700 (PDT) {}name: "Contract_Insight_Report_marzatem_JPMC.xlsm"size: 265892023type: "application/vnd.ms-excel.sheet.macroenabled.12"webkitRelativePath: ""__proto__: File

demo.js:246 {
  "SheetNames": [
    "Sheet1"
  ],
  "Sheets": {
    "Sheet1": {}
  }
}

I'm afraid the string limit increase won't help as this is 10X larger than the file @sigod linked to. 

If my need is only to read the data itself (no cell meta data) does the Pro version offer a streaming solution?  @SheetJSDev

**Comment by pckhoi** [comment]
Is it possible to store the entire file in a typed array instead of string? In Chrome maximum buffer length is `2 ** 31 - 2 ** 12 = 2147479552` bytes. Firefox's is double that. That should be more than enough.

**Comment by Kaiido** [comment]
I didn't check thoroughly the sources yet, and I don't have time to work on this myself right now, but from what I understand you are using `JSZipSync` to first unzip the xlsx file.   
The sheets are then accessible in the object (`obj`) returned by `JSZipSync` as `obj["files"]["xl/worksheets/sheetN.xml"]` which is actually yet an other compressed file.  
Calling the `.getContent()` method of this object will return an Uint8Array representing the xml file.   

So theoretically, all it takes to have stream-reading and huge files support is to hook here a stream-xml-parser (it seems there are many out-there), and to read that xml file by chunks (which can be done efficiently in a browser using a TextDecoder).  
Am I missing something else here that would prevent implementing this more than necessary feature?

**Comment by Kaiido** [comment]
Toady is holiday here and I had time to make a simple Proof-Of-Concept of my idea in https://github.com/SheetJS/sheetjs/issues/792#issuecomment-785733626

You can see it at https://xml-stream-parsing-poc.glitch.me/

Sorry it is very fast written since I lack time, and it might make your eyes get out of their orbit seeing how bad my xlsx parser is (I really just wanted to somehow log the values in my sheet), but it at least shows that it should be possible to implement such an xml stream parser in your library too.   
Using that I was able to parse 60MB+ files in Chrome, while SheetJS fails to read these.

**Comment by kisaragi99** [comment]
Need help.
I have the same error message as jensJJ had.
Tried to parse `calendar_stress_test` file. (74mb, 1 million rows)
https://github.com/SheetJS/sheetjs/issues/61#issuecomment-47676371
Converted it to xlsx.

Got this error.

![image](https://user-images.githubusercontent.com/58854608/146205982-300c5c42-c2b5-439f-8e8b-7109edd043d5.png)

If im not mistaken - the issue with V8 max string length is obsolete and the size of this file (740mb) should be fine?


```
Archive:  calendar_stress_test_copy.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
     1348  01-01-1980 00:00   [Content_Types].xml
      733  01-01-1980 00:00   _rels/.rels
      831  01-01-1980 00:00   xl/_rels/workbook.xml.rels
     1393  01-01-1980 00:00   xl/workbook.xml
      383  01-01-1980 00:00   xl/sharedStrings.xml
     7646  01-01-1980 00:00   xl/theme/theme1.xml
     2285  01-01-1980 00:00   xl/styles.xml
572078353  01-01-1980 00:00   xl/worksheets/sheet1.xml
    38292  01-01-1980 00:00   docProps/thumbnail.jpeg
      792  01-01-1980 00:00   docProps/app.xml
168383066  01-01-1980 00:00   xl/calcChain.xml
      593  01-01-1980 00:00   docProps/core.xml
---------                     -------
740515715                     12 files

```


### But, it worked perfectly in Firefox - approx 2 minutes.
(No browser extensions enabled in both chrome and firefox)
